### PR TITLE
Enable xapian in mailman-web

### PR DESCRIPTION
**Pull request by pgeorgi**

### CODE REVIEW

File: web/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.6-alpine3.7
+FROM alpine:latest
```
Comment (maxking) [inline web/Dockerfile]: Why do we need to switch to `alpine` base image instead of using the one we are currently using?

File: web/Dockerfile [hunk 2]
```diff
@@ -26,6 +27,7 @@ RUN set -ex \
 		dj-database-url \
 		mysqlclient \
 		typing \
+	&& python3 -m pip install git+https://github.com/notanumber/xapian-haystack.git \
```
Comment (maxking) [inline web/Dockerfile]: Not a big fan of installing packages directly from Github, is this package not published in PyPI?

File: web/Dockerfile [hunk 3]
```diff
@@ -12,11 +12,12 @@ COPY docker-entrypoint.sh /usr/local/bin/
 # rights for management script
 RUN set -ex \
 	&& apk add --no-cache --virtual .build-deps gcc libc-dev linux-headers \
-	   	postgresql-dev mariadb-dev \
+		postgresql-dev mariadb-dev python3-dev git \
```
Comment (maxking) [inline web/Dockerfile]: Any reason we need `git` in this image?

Comment (pgeorgi) [reply web/Dockerfile]: For the github download above. If we go for another source, it can be dropped.

File: web/Dockerfile (hunk 2 above)
Comment (pgeorgi) [reply web/Dockerfile]: I went by the documentation on that repo: https://github.com/notanumber/xapian-haystack#installation. It exists on PyPI, so I'll update the routine to use https://pypi.org/project/xapian-haystack/ instead.

File: web/Dockerfile (hunk 1 above)
Comment (pgeorgi) [reply web/Dockerfile]: python:* packages build python from source and into /usr/local. That makes it harder to reuse alpine packages that interface with python (eg. xapian's python bindings that haystack then uses).
By going for "pure" alpine and its packages, we have a single source of binaries.

Comment (maxking) [reply web/Dockerfile]: It sounds reasonable, but let's not use the `latest` tag, I'd much prefer that we pick whatever is the latest version (3.7?).

**Comment by maxking** [comment]
Can you also rebase off the master? I bumped Django and MailmanClient in master.

**Comment by maxking** [comment]
Thanks very much for your fixes!

I am gonna try out this change locally and then merge it. This is first time we are bumping to alpine 3.8, so I just want to be extra cautious since a few people do use the rolling releases generated daily :)

**Pull request merged**